### PR TITLE
[channel-client] Include src and tests directories in npm package

### PR DESCRIPTION
**Pull request by kerzhner**

### CODE REVIEW

File: packages/channel-client/package.json [hunk 1]
```diff
@@ -26,6 +26,10 @@
     "ts-jest": "25.0.0",
     "typescript": "3.7.5"
   },
+  "files": [
+    "lib/src",
+    "lib/tests"
```
Comment (andrewgordstewart) [inline packages/channel-client/package.json:31]: Why would you include the tests?

Comment (kerzhner) [reply packages/channel-client/package.json:31]: https://github.com/statechannels/monorepo/blob/39d32bd65bd2aabd27d831eddb25035dc368fc5a/packages/channel-client/src/index.ts#L6

Comment (kerzhner) [reply packages/channel-client/package.json:31]: Tests are imported in `index.ts` which is the package entrypoint.

Comment (andrewgordstewart) [reply packages/channel-client/package.json:31]: Can we include just that file?

It's besides the point of this PR, but it's strange to me that the `tests` folder of the `channel-client` package must export a mock `ChannelProvider`.

Comment (kerzhner) [reply packages/channel-client/package.json:31]: We probably can include just that fille.

Comment (kerzhner) [reply packages/channel-client/package.json]: 58fec193a

**Pull request merged**